test/support/test_validator.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Support.TestValidator do
+  @moduledoc """
+  A MessageValidator implementation. If no stream_key is assigned, all streams
+  will be allowed.
+  """
+
+  defstruct stream_key: nil
+end
+
+defimpl Membrane.RTMP.MessageValidator, for: Support.TestValidator do
+  alias Membrane.RTMP.Messages
+
+  @impl true
+  def validate_publish(%Support.TestValidator{stream_key: nil}, _message),
+    do: {:ok, "stream allowed"}
+
+  def validate_publish(%Support.TestValidator{stream_key: accepted_key}, %Messages.Publish{
+        stream_key: stream_key
+      }) do
+    if stream_key == accepted_key do
+      {:ok, "correct stream key"}
+    else
+      {:error, "wrong stream key"}
+    end
+  end
+
+  @impl true
+  def validate_release_stream(_impl, _message), do: {:ok, "release stream success"}
```

Comment:
nitpick: can you do the `defdelegate` here so other people who would like to get inspired by the tests can look it up?

Reply:
As far as I know, you cannot delegate in this case as the MessageValidator.Default implements those functions in its protocol implementation only, they do not appear as functions of the module... 🙈

Reply:
version bumped

Reply:
Please bump to `0.12.0` instead since we are changing API.